### PR TITLE
PRT - fix_jsonrpc_ids_parsing_on_error_handling

### PR DESCRIPTION
**Pull request by candostyavuz**

### CODE REVIEW

File: protocol/chainlib/jsonRPC.go [hunk 1]
```diff
@@ -484,6 +484,18 @@ func (cp *JrpcChainProxy) SendNodeMsg(ctx context.Context, ch chan interface{},
 			return nil, "", nil, utils.LavaFormatError("jsonRPC error", err, utils.Attribute{Key: "GUID", Value: ctx})
 		}
 		replyMsg = *replyMessage
+
+		reqId, idErr := rpcInterfaceMessages.IdFromRawMessage(nodeMessage.ID)
```
Comment (ranlavanet) [inline protocol/chainlib/jsonRPC.go:488]: let's create a common method that we use and add it to the error handler class. for both tendermint and jsonrpc

**Review by ranlavanet — CHANGES_REQUESTED** [review]

File: protocol/chainlib/jsonRPC.go (hunk 1 above)
Comment (candostyavuz) [reply protocol/chainlib/jsonRPC.go]: fixed in: 92fb9142c18e1e95113212df7301f824f46e1e14

File: protocol/chainlib/jsonRPC.go [hunk 2]
```diff
@@ -484,6 +484,11 @@ func (cp *JrpcChainProxy) SendNodeMsg(ctx context.Context, ch chan interface{},
 			return nil, "", nil, utils.LavaFormatError("jsonRPC error", err, utils.Attribute{Key: "GUID", Value: ctx})
 		}
 		replyMsg = *replyMessage
+
+		err := cp.HandleIDMismatchError(nodeMessage.ID, replyMessage.ID)
```
Comment (ranlavanet) [inline protocol/chainlib/jsonRPC.go:488]: lets change the name of this to ValidateRequestAndResponseIds 
so its clear its a validation.

**Review by ranlavanet — CHANGES_REQUESTED** [review]

File: protocol/chainlib/jsonRPC.go (hunk 2 above)
Comment (candostyavuz) [reply protocol/chainlib/jsonRPC.go]: fixed: 3e029e715b1a279a51f3bf50120ead79aaa4cffe

**Review by ranlavanet — APPROVED** [review]

**Pull request merged**